src/transport/SessionHolder.h

```diff
@@ -68,6 +68,8 @@ class SessionHolderWithDelegate : public SessionHolder
 {
 public:
     SessionHolderWithDelegate(SessionReleaseDelegate & delegate) : mDelegate(delegate) {}
+    SessionHolderWithDelegate(SessionHolder holder, SessionReleaseDelegate & delegate) : SessionHolder(holder), mDelegate(delegate)
```

Comment:
Should this pass by value or by rvalue reference, for the holder param?

Reply:
Done.